### DISCUSSION TITLE
Adding a member to a team always gives them the role of "maintainer"

### DISCUSSION THREAD

**Discussion by dranidis** [community/community#140675] [Enterprise] [answered]
### Select Topic Area

Question

### Body

I am using the gh api to add members to teams of my organization.
The documentation https://docs.github.com/en/rest/teams/members?apiVersion=2022-11-28#add-or-update-team-membership-for-a-user states:

**Body parameters**
role string
The role that this user should have in the team.
Default: member
Can be one of: member, maintainer

I want to assign the role of a `member` but whatever I do I fail.

The following
```
gh api \
   --method PUT \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
   /orgs/$ORG/teams/$TEAM/memberships/$MEMBER -f "role=member" 
```
returns
```
{
  "state": "active",
  "role": "maintainer",
  "url": "https://api.github.com/organizations/........"
}
```

I even tried removing the ` -f "role=member"` since the documentation states that the default is `member`, but the role is still `maintainer`. 

Any help or clarification is appreciated!

**Comment by Topromm** [comment]
Double-check that the team doesn't contain any settings or permissions that could be overriding what you're trying to do. Sometimes orgs have rules that can automatically bump certain users to "maintainer."

You can also try debugging with a simpler API call just to see what it returns without passing any role (since "member" is the default).

Also, try updating to the latest version of the GitHub CLI if you haven't already. Could be something small that's been fixed in newer releases?

If none of that works and you think it could be a bug, I would turn to GitHub support or open a ticket.

**Comment by dranidis** [comment]
Hi @Topromm, 

thanks for the response.

Concerning settings and permissions, I did not find anything about default roles in the team settings or organization settings. Can you please tell me where are the settings you are referring to?

As for the simpler API call, I have already tried passing no role. The answer is still "maintainer"

I am currently using gh 2.56. The latest is 2.58. Hard to think, that this might be the reason, but I will check.

**Comment by dranidis** [comment, marked as answer]
Problem solved. The role is "maintainer" by default if the person is an owner. It does not matter if you provide a role explicitly. For non owners the role is indeed member.
